relay-sampling/src/lib.rs

```diff
@@ -1,51 +1,57 @@
+//!
+//! 1. **Trace sampling** ensures that either all transactions of a trace are sampled or none. Rules
+//!    have access to information in the [`DynamicSamplingContext`].
+//! 2. **Transaction sampling** does not guarantee complete traces and instead applies to individual
+//!   transactions. Rules have access to the full data in transaction events.
 //!
 //! # Components
 //!
 //! The sampling system implemented in Relay is composed of the following components:
-//! - [`DynamicSamplingContext`]: a struct that contains the trace information.
-//! - [`SamplingRule`]: a rule that is matched against [`Event`] or [`DynamicSamplingContext`] that
-//! can contain a [`RuleCondition`] for expressing predicates on the incoming payload.
-//! - [`SamplingMatch`]: the result of the matching of one or more [`SamplingRule`].
+//!
+//! - [`DynamicSamplingContext`]: a container for information associated with the trace.
+//! - [`SamplingRule`](crate::config::SamplingRule): a rule that is matched against incoming data.
+//!   It specifies a condition that acts as predicate on the incoming payload.
+//! - [`SamplingMatch`](crate::evaluation::SamplingMatch): the result of the matching of one or more
+//!   rules.
 //!
 //! # How It Works
-//! - The incoming [`Event`] and optionally [`DynamicSamplingContext`] are received by Relay.
-//! - Relay fetches the [`SamplingConfig`] of the project to which the [`Event`] belongs and (if exists) the
-//! [`SamplingConfig`] of the root project of the trace.
-//! - The [`SamplingConfig`]s are merged together and the matching algorithm in
-//! [`SamplingMatch::match_against_rules`] is executed.
-//! - The sampling algorithm will go over each [`SamplingRule`] and compute either a factor or
-//! sample rate based on the [`SamplingValue`] of the rule.
-//! - The [`SamplingMatch`] is finally returned containing the final `sample_rate` and some additional
-//! data that will be used in `relay_server` to perform the sampling decision.
 //!
-//! # Sampling Determinism
-//! The concept of determinism is extremely important for sampling. We want to be able to make the
-//! a deterministic sampling decision for a wide variety of reasons, including:
-//! - Across a **chain of Relays** (e.g., we don't want to drop an event that was retained by a previous
-//! Relay and vice-versa).
-//! - Across **transactions of the same trace** (e.g., we want to be able to sample all the transactions
-//! of the same trace, even though some exceptions apply).
+//! - The incoming data is received by Relay.
+//! - Relay resolves the [`SamplingConfig`] of the project to which the incoming data belongs.
+//!   Additionally, it tries to resolve the configuration of the project that started the trace, the
+//!   so-called "root project".
+//! - The two [`SamplingConfig`]s are merged together and dynamic sampling evaluates a result. The
+//!   algorithm goes over each rule and compute either a factor or sample rate based on the
+//!   value of the rule.
```

Comment:
Mentioning sampling config for the first, configuration for the second, and then referring to the sampling config for both is a bit confusing. What do you think of the following?
```suggestion
//! - Relay resolves the [`SamplingConfig`] of both the project to which the incoming data belongs
//!   and the project that started the trace, the so-called "root project".
//! - The two [`SamplingConfig`]s are merged together and dynamic sampling evaluates a result. The
//!   algorithm goes over each rule and compute either a factor or sample rate based on the
//!   value of the rule.
```